accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -679,6 +683,19 @@ If no type is specified in a catch clause, then the default type of the error
 variable is `Object`, instead of `dynamic` as was the case in pre-null safe
 Dart.
 
+#### Spread element typing
+
+In a collection literal in Dart before null-safety, the inferred element
+type of a spread element of the form `...?e` where `e` has static type
+`Null` is `Null`, and so are the inferred key type and value type.
+
+With null-safety, when the static type of `e` is `Null` or a potentially
```

Comment:
First off, I'm confused by the phrasing here.  I think you're trying to cover two cases: `...?e` where `e` has certain types, and `...e` where `e` has certain types?  As a textual thing, I'd suggest to split these out into two separate paragraphs for clarity, even at the expense of some redundancy.

Second, I'm not sure this covers all the cases.  The first sentence covers `...?e` where `e` has type `Null`, or `X extends Null` or `X & Null`, and I think nothing else.  The second sentence covers `...e` when `e` has type `Never` or `X extends Never` or `X & Never`.  I think the second syntax with the first set of types is an error?  But shouldn't the second set of types with the first syntax also be valid and have type `Never`?  I think it's specified as a warning, but not an error, right?

Reply:
Indeed, fixed that.